### PR TITLE
Fix CI failing

### LINKED ISSUE #3953: GitHub Action fails
**Describe the bug**
GitHub Action fails with the latest version, 23.10.0. If I specify a version (e.g., 23.9.0), it works fine.
I have been using the GitHub Action snippet as described on [this page](https://black.readthedocs.io/en/stable/integrations/github_actions.html). 

**To Reproduce**

For example, take this code:

```
name: Lint

on: [push, pull_request]

jobs:
  lint:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: psf/black@stable
        with:
          options: "--check --verbose"
```

The resulting error is:
The GitHub Action raises an error. 
스크린샷 2023-10-17 오후 5 40 45

**Expected behavior**
If I specify a _black_ version like this, it works fine.
```
name: Lint

on: [push, pull_request]

jobs:
  lint:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: psf/black@23.9.1
        with:
          options: "--check --verbose"
```

**Environment**

- Black's version: psf/black@stable (or 23.10.0)
- OS and Python version: GitHub Action

**Additional context**

### PR DESCRIPTION
**Pull request by kiyoon**
### Description

Fixes #3953 

### Checklist - did you ...

- [x] Add an entry in `CHANGES.md` if necessary?
- [ ] Add / update tests if necessary?
- [ ] Add new / update outdated documentation?

<!-- Just as a reminder, everyone in all psf/black spaces including PRs
     must follow the PSF Code of Conduct (link below).

     Finally, once again thanks for your time and effort. If you have any
     feedback in regards to your experience contributing here, please
     let us know!

     Helpful links:

      PSF COC: https://www.python.org/psf/conduct/A

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
Thanks! Could you add a NEWS entry since this fixes a user-visible issue?

We'll probably cut a hotfix release for this.

**Comment by kiyoon** [comment]
By NEWS entry, you mean in CHANGELOG?

**Review by cooperlees — APPROVED** [review]

**Comment by cooperlees** [comment]
I think we need to fix the pre-commit checks elsewhere ourselves if I'm reading them correctly?

File: CHANGES.md [hunk 1]
```diff
@@ -47,6 +47,12 @@
 <!-- Major changes to documentation and policies. Small docs changes
      don't need a changelog entry. -->
 
+## 23.10.1
+
+### Integrations
+
+- Fix the action failing when Black check doesn't pass (#3957)
```
Comment (JelleZijlstra) [inline CHANGES.md:54]: This should go into the "Unreleased" section above. We'll set this to 23.10.1 when we make the release.

Comment (kiyoon) [reply CHANGES.md]: Done!

Comment (JelleZijlstra) [reply CHANGES.md]: Thanks!

**Comment by kiyoon** [comment]
You can see that this is running with a failing file here:

https://github.com/kiyoon/blacktest/actions/runs/6578198222

**Comment by csalerno-asml** [comment]
Hey, sorry for the issue, I noticed today the comment in the original PR: https://github.com/psf/black/pull/3914#issuecomment-1771204213

My proposal was to add ` && exit_code=0 || exit_code=$?` after `out=...`, but this is also fine. Thanks for the PR and sorry for the bug!

**Pull request merged**